### PR TITLE
#361: Highcharts in dashboard theme

### PR DESCRIPTION
**Pull request by divino**
TODO: The result of the co-occurence doesn't match the data I expected. Might need a fix on the ml-highcharts-ng. This affects the gender vs eyeColor chart. #361

Correction: Data doesn't match because of the resultLimit.

### CODE REVIEW

File: app/themes/dashboard/ui/app/app.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    'app.root',
+    'app.search',
+    'app.user',
+    'ui.bootstrap',
+    'ui.router',
+    'ui.tinymce',
+    'ngToast'
+  ]);
+
+}());
```
Comment (grtjn) [inline app/themes/dashboard/ui/app/app.js]: looks like this doesn't contain actual changes?

File: slushfile.js [hunk 2]
```diff
@@ -341,6 +341,26 @@ function configRoxy() {
...
         '          <range-value-positions>false</range-value-positions>\n' +
         '          <invalid-values>reject</invalid-values>\n' +
+        '        </range-path-index>\n' +
+        '        <range-path-index>\n' +
+        '          <scalar-type>string</scalar-type>\n' +
+        '          <collation>http://marklogic.com/collation/codepoint</collation>\n' +
+        '          <path-expression>eyeColor</path-expression>\n' +
+        '          <range-value-positions>false</range-value-positions>\n' +
+        '          <invalid-values>reject</invalid-values>\n' +
+        '        </range-path-index>\n' +
```
Comment (grtjn) [inline slushfile.js]: Ah wait, we should drop the element index in favor of the path index, and adjust all.xml accordingly, as that works better with the recent xml/json edit features..

File: app/themes/dashboard/ui/app/route/routes.js [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+        controllerAs: 'ctrl'
+      })
+      .state('root.login', {
+        url: '/login?state&params',
+        templateUrl: 'app/login/login-full.html',
+        controller: 'LoginFullCtrl',
+        controllerAs: 'ctrl'
+      });
+  }
+}());
```
Comment (grtjn) [inline app/themes/dashboard/ui/app/route/routes.js]: Not entirely sure, was anything actually changed in here? Bit hard to compare without left side..

**Comment by grtjn** [comment]
Looks promising at first glance!

**Comment by divino** [comment]
I added the module of the landing which doesn’t exist on other themes.

Regards,

Dino Bagayaua
Senior Field Engineer
MarkLogic Corporation
Dino.Bagayaua@marklogic.com
Cell: +63-998-569-5332
www.marklogic.com

**Comment by divino** [comment]
Landing here now have it’s own controller.

Regards,

Dino Bagayaua
Senior Field Engineer
MarkLogic Corporation
Dino.Bagayaua@marklogic.com
Cell: +63-998-569-5332
www.marklogic.com

File: app/themes/dashboard/ui/app/route/routes.js (hunk 3 above)
Comment (divino) [inline app/themes/dashboard/ui/app/route/routes.js]: Landing here now have it’s own controller.

File: app/themes/dashboard/ui/app/app.js (hunk 1 above)
Comment (divino) [inline app/themes/dashboard/ui/app/app.js]: I added the module of the landing which doesn’t exist on other themes.

**Comment by grtjn** [comment]
Second review: still looks very promising. I'll reach out to Ryan D about ml-highcharts-ng (not your concern), and i'm thinking there should be a default theme landing ctrl (not your concern either).

Still need to run a quick test, will try to look into that later tonight..

**Comment by grtjn** [comment]
I'd like to make a few little tweaks, but a very good start nonetheless. Thnx, merging to be able to pile up some changes myself..

**Pull request merged**